libs/wss/src/lib.rs

```diff
@@ -0,0 +1,134 @@
+use std::io::{BufReader, Read, Seek, Write};
+
+use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+
+mod compression;
+mod error;
+mod wav;
+
+pub use compression::Compression;
+pub use error::Error;
+
+pub struct Wss {
+    compression: Compression,
+    format: u16,
+    channels: u16,
+    sample_rate: u32,
+    bytes_per_second: u32,
+    block_align: u16,
+    bits_per_second: u16,
+    output_size: u16,
+    data: Vec<u8>,
+}
+
+impl Wss {
+    /// Read a WSS file from the input
+    ///
+    /// # Errors
+    /// [`std::io::Error`] if an IO error occurs
+    /// [`Error::UnsupportedFileType`] if the file is not a WSS file
+    /// [`Error::InvalidCompressionValue`] if the compression value is invalid
+    pub fn read<I: Read + Seek>(input: &mut I) -> Result<Self, Error> {
+        let mut reader = BufReader::new(input);
+
+        let mut buffer = [0; 4];
+        reader.read_exact(&mut buffer)?;
+        if &buffer != b"WSS0" {
+            return Err(Error::UnsupportedFileType(
+                String::from_utf8_lossy(&buffer).to_string(),
+            ));
+        }
+
+        let compression = reader.read_u32::<LittleEndian>()?;
```

Comment:
compression is actually a single byte, and 3 bytes padding.
There is no guarantee for the padding to be zeroed